tgz.go

```diff
@@ -67,6 +103,10 @@ func (a *Archive) AddFileByContent(src []byte, dest string) error {
 }
 
 func (a *Archive) Close() {
+	close(a.WriteQueue)
+
+	<-a.Finished
```

Comment:
Yes that is what I was saying. Just making sure I understood why we were doing this